src/scss/_functions.scss

```diff
@@ -237,34 +237,42 @@
 /// Returns a text color based on the background and
 /// an opacity percentage the color should appear at
 ///
-/// @param {Color} $background - the hex color of the background the text will appear on
+/// @param {Color} $background - the palette name or hex of the background the text will appear on
 /// @param {Number} $opacity [100] - the opacity percentage the text color should appear at
-/// @param {Boolean} $warnings [true] - whether this function should throw WCAG contrast check warnings/errors
-@function oColorsGetTextColor($backgroundd, $opacity: 90, $warnings: true) {
-	$background: if(type-of($backgroundd) == 'string', oColorsByName($backgroundd), $backgroundd);
-
-	@if $background == null or type-of($background) != color {
+/// @param {Number|Null} $minimum-contrast [4.5] - the minimum contrast ratio between the background and the returned text color. If the contrast ratio is lower an error is thrown. WCAG 2.1 level AA requires a contrast ratio of at least 4.5:1 for normal text and 3:1 for large text. Set to `null` to not throw an error.
+@function oColorsGetTextColor($background, $opacity: 90, $minimum-contrast: 4.5) {
+	// Get background hex if palette colour name has been given.
+	$background-name: $background;
+	$background: if(type-of($background) == 'string', oColorsByName($background), $background);
+
+	// Validate argument types.
+	@if type-of($background) != color {
 		@return _error("'#{inspect($background)}' is not a valid color. To get a text color, please supply a valid hex code or color name for the background color'");
 	}
-
-	$percentage: $opacity;
-	$baseColor: _oColorsGetTextBase($background);
-	$textColor: oColorsMix($baseColor, $background, $percentage);
-	$testContrast: oColorsCheckContrast($textColor, $background, false);
-
-	@if not $testContrast {
-		$baseColor: if($baseColor == 'black', 'white', 'black');
-		$textColor: oColorsMix($baseColor, $background, $percentage);
-		$testContrast: oColorsCheckContrast($textColor, $background, false);
+	@if type-of($opacity) != 'number' {
+		@return _error("'#{inspect($opacity)}' is not a valid opacity, set to a number.'");
 	}
-
-	@if not $testContrast and $warnings == true {
-		@return _error("The combination of #{$opacity}% #{$baseColor} on #{$background} does not pass WCAG guidelines for color contrast.");
+	@if $minimum-contrast != null and type-of($minimum-contrast) != 'number' {
```

Comment:
what if we made it easier on the user and accept an enum of like `$text-size: 'large'` or '`$text-size: 'normal'` or `$text-size: 'null'` so that they don't have to know the contrasts?

Reply:
I'd like to keep the mixin flexible for use in different contexts, like if a user wanted to meet higher AAA standards or if they insist on better or worse contrast for some other, maybe even stylistic reason. But I think you're right to call out it could be more helpful for our users... Two alternate ideas, what if:

1. we in the error we added a link and an excerpt from WCAG21 giving the user the values?
2. what you said but allow a contrast number or a string `$minimum-contrast: 'normal'` 

I think (1) could work because it's hopefully an argument our users can ignore unless there's an error.

Reply:
do we really want people to be able to insist on specific contrasts for stylistic reasons, when this is about accessibility?

what if it could be a map like `$minimum-contrast: (level: 'aaa', text-size: 'normal')`.

it depends on what this feature (the `$minimum-contrast` option) is for. is it meant to be for people to be able to meet the correct wcag standard for their site?
or is it for people to be able to choose whether or not they want their site to be useable for people with disabilities?

Reply:
I like the idea of a map! That'll let the user be more strict if they want to be, or `null` to remove the check. E.g. for:
>Text or images of text that are part of an inactive user interface component, that are pure decoration, that are not visible to anyone, or that are part of a picture that contains significant other visual content, have no contrast requirement.

Reply:
having `null` as an option for decorative i prefer a lot, compared to giving the opportunity to set their `$minimum-contrast` to exactly the magic number that doesn't fail, hahaha.

the map gives makes it easier for them to do the right thing, and the code explains itself :)
seeing as there are so few options it could be a map or a string like: `aa-normal`, `aaa-large`

Reply:
💯